client/src/components/ElicitationModal.tsx

```diff
@@ -0,0 +1,181 @@
+
+      for (const [fieldName, fieldValue] of Object.entries(dataObj)) {
+        const fieldSchema = schema.properties[fieldName];
+        if (
+          fieldSchema &&
+          fieldSchema.format === "email" &&
+          typeof fieldValue === "string"
+        ) {
+          if (!validateEmailFormat(fieldValue)) {
+            setValidationError(`Invalid email format: ${fieldName}`);
+            return false;
+          }
+        }
+      }
+    }
+
+    return true;
+  };
+
+  const handleAccept = () => {
+    try {
+      if (!validateFormData(formData, request.requestedSchema)) {
+        return;
+      }
+
+      const ajv = new Ajv();
+      const validate = ajv.compile(request.requestedSchema);
+      const isValid = validate(formData);
+
+      if (!isValid) {
+        const errorMessage = ajv.errorsText(validate.errors);
+        setValidationError(errorMessage);
+        return;
+      }
+
+      request.resolve({
+        action: "accept",
+        content: formData as Record<string, unknown>,
+      });
+      onClose();
+    } catch (error) {
+      setValidationError(
+        error instanceof Error ? error.message : "Validation failed",
+      );
+    }
+  };
+
+  const handleReject = () => {
+    request.resolve({ action: "reject" });
+    onClose();
+  };
+
+  const handleCancel = () => {
+    request.resolve({ action: "cancel" });
+    onClose();
+  };
+
+  const schemaTitle = request.requestedSchema.title || "Information Request";
+  const schemaDescription = request.requestedSchema.description;
+
+  return (
+    <Dialog open={true} onOpenChange={handleCancel}>
+      <DialogContent className="max-w-2xl max-h-[80vh] overflow-y-auto">
+        <DialogHeader>
+          <DialogTitle>{schemaTitle}</DialogTitle>
+          <DialogDescription className="space-y-2">
+            <span className="block">{request.message}</span>
+            {schemaDescription && (
+              <span className="block text-sm text-muted-foreground">
+                {schemaDescription}
+              </span>
+            )}
+          </DialogDescription>
+        </DialogHeader>
+
+        <div className="py-4">
+          <DynamicJsonForm
+            schema={request.requestedSchema}
+            value={formData}
+            onChange={(newValue: JsonValue) => {
+              setFormData(newValue);
+              setValidationError(null);
+            }}
+          />
+
+          {validationError && (
+            <div className="mt-4 p-3 bg-red-50 dark:bg-red-900/20 border border-red-200 dark:border-red-800 rounded-md">
+              <div className="text-sm text-red-600 dark:text-red-400">
+                <strong>Validation Error:</strong> {validationError}
+              </div>
+            </div>
+          )}
+        </div>
+
+        <DialogFooter className="gap-2">
+          <Button variant="outline" onClick={handleCancel}>
+            Cancel
+          </Button>
+          <Button variant="outline" onClick={handleReject}>
```

Comment:
`reject` has been reverted to `decline`. It was slipped into the documentation at the last minute, but not the schema. When this was realized, the team decided to revert rather than make a post-release update to the schema.

```suggestion
          <Button variant="outline" onClick={handleDecline}>
```